clang/test/CXX/drs/cwg1884.cpp

```diff
@@ -0,0 +1,548 @@
+class g;
+// since-cxx20-error@-1 {{redefinition of 'g' as different kind of symbol}}
+//   since-cxx20-note@cwg1884_A.cppm:49 {{previous definition is here}}
+template <typename>
+void g(int);
+// since-cxx20-error@-1 {{redefinition of 'g' as different kind of symbol}}
+//   since-cxx20-note@cwg1884_A.cppm:49 {{previous definition is here}}
+template <typename, typename>
+int g;
+// since-cxx20-error@-1 {{redefinition of 'g' as different kind of symbol}}
+//   since-cxx20-note@cwg1884_A.cppm:49 {{previous definition is here}}
+template <typename T>
+int g<T, int>;
+// since-cxx20-error@-1 {{no variable template matches partial specialization}}
+template <typename>
+using g = int;
+// since-cxx20-error@-1 {{redefinition of 'g' as different kind of symbol}}
+//   since-cxx20-note@cwg1884_A.cppm:49 {{previous definition is here}}
+template <typename>
+concept g = true;
+// since-cxx20-error@-1 {{redefinition of 'g' as different kind of symbol}}
+//   since-cxx20-note@cwg1884_A.cppm:49 {{previous definition is here}}
+
+
+// template <typename>
+// void h(int);
+
+int h;
+// since-cxx20-error@-1 {{redefinition of 'h' as different kind of symbol}}
+//   since-cxx20-note@cwg1884_A.cppm:51 {{previous definition is here}}
+void h(); // #cwg1884-function-f
+// @-1 OK, a non-corresponding overload
+enum Eh {
+  h
+  // FIXME-since-cxx20-error@-1 {{redefinition of 'h'}}
+  //   FIXME-since-cxx20-note@cwg1884_A.cppm:51 {{previous definition is here}}
+};
+namespace h {} // #cwg1884-namespace-h
+// FIXME-since-cxx20-error@-1 {{redefinition of 'h' as different kind of symbol}}
+//   FIXME-since-cxx20-note@cwg1884_A.cppm:51 {{previous definition is here}}
+struct h;
+// since-cxx20-error@-1 {{redefinition of 'h' as different kind of symbol}}
+//   since-cxx20-note@#cwg1884-namespace-h {{previous definition is here}}
+class h;
+// since-cxx20-error@-1 {{redefinition of 'h' as different kind of symbol}}
+//   since-cxx20-note@#cwg1884-namespace-h {{previous definition is here}}
+template <typename>
+class h;
+// FIXME-since-cxx20-error@-1 {{redefinition of 'h' as different kind of symbol}}
+//   FIXME-since-cxx20-note@cwg1884_A.cppm:51 {{previous definition is here}}
+template <typename, typename>
+int h;
+// since-cxx20-error@-1 {{redefinition of 'h' as different kind of symbol}}
+//   since-cxx20-note@#cwg1884-namespace-h {{previous definition is here}}
+template <typename T>
+int h<T, int>;
+// since-cxx20-error@-1 {{no variable template matches specialization; did you mean to use 'h' as function template instead?}}
+template <typename>
+using h = int;
+// since-cxx20-error@-1 {{redefinition of 'h' as different kind of symbol}}
+//   since-cxx20-note@cwg1884_A.cppm:51 {{previous definition is here}}
+template <typename>
+concept h = true;
+// since-cxx20-error@-1 {{redefinition of 'h' as different kind of symbol}}
+//   since-cxx20-note@#cwg1884-function-f {{previous definition is here}}
+
+
+// template <typename, typename>
+// int i;
+
```

Comment:
is that the thing we don't support? There should be a fixme?

Reply:
Hopefully the new structure of the test should make it less confusing.